src/base/net/private/downloadhandlerimpl.cpp

```diff
@@ -66,7 +66,7 @@ DownloadHandlerImpl::DownloadHandlerImpl(Net::DownloadManager *manager, const Ne
 
 DownloadHandlerImpl::~DownloadHandlerImpl()
 {
-    delete m_reply;
+    m_reply->abort();
```

Comment:
Isn't `abort()` called by the destructor?
The "reply" can be still not assigned here.

Reply:
>The "reply" can be still not assigned here.

In `DownloadHandlerImpl::assignNetworkReply()` there is `m_reply->setParent(this);` and this is the only way to assign `m_reply`, right?

>Isn't abort() called by the destructor?

Following above, then we don't need to delete m_reply in destructor, we only need to abort it.

Reply:
>Following above, then we don't need to delete m_reply in destructor, we only need to abort it.

I don't know how you came to that conclusion...
You can't just call `abort()` here since m_reply can be nullptr. But in the case it is assigned (i.e. not null) should we really call `abort` explicitly? Isn't it called in QNetworkReply destructor?

Reply:
>You can't just call abort() here since m_reply can be nullptr. 

I missed that.

>But in the case it is assigned (i.e. not null) should we really call abort explicitly? Isn't it called in QNetworkReply destructor?

Seem yes, then I suppose `DownloadHandlerImpl::~DownloadHandlerImpl()` can be removed.